packages/react/src/useActionForm.ts

```diff
@@ -0,0 +1,324 @@
+import type {
+  ActionFunction,
+  AnyModelManager,
+  DefaultSelection,
+  GadgetRecord,
+  GlobalActionFunction,
+  Select,
+} from "@gadgetinc/api-client-core";
+import { camelize } from "@gadgetinc/api-client-core";
+import { useCallback, useEffect, useRef } from "react";
+import type { DeepPartial, FieldErrors, FieldValues, UseFormProps, UseFormReturn } from "react-hook-form";
+import { useForm } from "react-hook-form";
+import { useApi } from "./GadgetProvider.js";
+import { useAction } from "./useAction.js";
+import { useFindBy } from "./useFindBy.js";
+import { useFindOne } from "./useFindOne.js";
+import { useGlobalAction } from "./useGlobalAction.js";
+import type { ActionHookState, ErrorWrapper, OptionsType } from "./utils.js";
+import { get, set } from "./utils.js";
+
+export * from "react-hook-form";
+
+/** Finds a given record from the backend database by either id or a `{[field: string]: value}` slug */
+const useFindExistingRecord = (
+  modelManager: AnyModelManager | undefined,
+  findBy: string | { [key: string]: any },
+  options: { select?: Record<string, any>; pause?: boolean }
+): [{ data?: GadgetRecord<any>; fetching: boolean; error?: ErrorWrapper }, () => void] => {
+  // for simplicity, we conditionally call either the findBy or findOne hook. this violates the rules of hooks, but is a LOT simpler than mounting and pausing both hooks and massaging the results of each together. this means we don't support the same form toggling from being for a record found by id and then later a record found by some other criteria, but that's a very rare use case. you can workaround this by adding a `key` prop to the component calling `useActionForm`, and having the value change when the find method changes, which will give different instance of the component and avoid the hook order changing.
+  if (modelManager && typeof findBy === "string") {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useFindOne(modelManager as any, findBy, options);
+  } else if (modelManager) {
+    const findByKey = Object.keys(findBy)[0];
+    const findByValue = Object.values(findBy)[0];
```

Comment:
nit: I'm not 100% sure if iteration order is guaranteed across browsers so maybe better to use `Object.entries` to get both key and value?